docs/data/toolpad/core/components/crud/crud.md

```diff
@@ -0,0 +1,356 @@
+---
+productId: toolpad-core
+title: Crud
+components: Crud, CrudProvider, List, Show, Create, Edit, CrudForm
+---
+
+# Crud
+
+<p class="description">The Crud component provides UIs to interact with data from any data source.</p>
+
+With the `Crud` component and its subcomponents you can easily generate pages where items from your data source can be listed in a table, shown individually in detail, or created and edited with forms. All with minimal configuration from a single data source definition.
+
+## Demo
+
+The `Crud` component automatically generates CRUD pages for your data source, automatically integrated with your routing solution. All that is required is a `dataSource` definition and a `rootPath` for the CRUD pages.
+
+{{"demo": "CrudBasic.js", "height": 600, "iframe": true}}
+
+The pages will be present in the following routes:
+
+- **List**: `/${rootPath}`
+- **Show**: `/${rootPath}/:id`
+- **Create**: `/${rootPath}/new`
+- **Edit**: `/${rootPath}/:id/edit`
+
+These default paths and other out-of-the-box settings can be overriden and configured in more detail by following the [advanced configuration](https://mui.com/toolpad/core/react-crud/#advanced-configuration) below.
+
+It is recommended to include the `dataSourceCache` prop in order to properly cache results from the data source query methods. See more in the section about [data caching](#data-caching) below.
+
+Optionally, additional configuration options can be provided such as `initialPageSize` for the paginated list of items, or `defaultValues` to set the initial form values when using a form to create new items.
+
+:::info
+Take a look at our examples for [Next.js](https://github.com/mui/toolpad/tree/master/examples/core/auth-nextjs/) and [Vite](https://github.com/mui/toolpad/tree/master/examples/core/vite/) for possible implementations on how the `Crud` component can be integrated with different routing solutions.
+:::
+
+## Data sources
+
+A **data source** is an object with a particular shape that must be used to configure the `Crud` component or its subcomponents.
+It includes:
+
+- A `fields` property, which extends the [MUI X Data Grid column definition](https://mui.com/x/react-data-grid/column-definition/) where different fields, header names and field types, among others, can be specified for your data.
+
+  An `id` field (string or number) is mandatory so that individual items can be identified.
```

Comment:
```suggestion
 - An `id` field (string or number) is mandatory so that individual items can be identified.
```

Reply:
These lines are kind of "under" the first point, that's why I structured them like this. I'll see if there's a better way to structure it.